File: programas/Python/unidad1/calculadora.py
# Función para sumar dos números
def sumar(a, b):
    return a + b

# Función para restar dos números
def restar(a, b):
    return a - b

# Función para multiplicar dos números
def multiplicar(a, b):
    return a * b

# Función para dividir dos números
def dividir(a, b):
    if b == 0:
        return "Error: No se puede dividir por cero"
    return a / b

# Menú de la calculadora
while True:
    print("Opciones:")
    print("Escribe 'sumar' para sumar dos números")
    print("Escribe 'restar' para restar dos números")
    print("Escribe 'multiplicar' para multiplicar dos números")
    print("Escribe 'dividir' para dividir dos números")
    print("Escribe 'salir' para salir de la calculadora")
    
    opcion = input(": ")

    if opcion == "salir":
        break

    if opcion in ("sumar", "restar", "multiplicar", "dividir"):
        num1 = float(input("Ingresa el primer número: "))
        num2 = float(input("Ingresa el segundo número: "))

        if opcion == "sumar":
            print("Resultado:", sumar(num1, num2))

        elif opcion == "restar":
            print("Resultado:", restar(num1, num2))

        elif opcion == "multiplicar":
            print("Resultado:", multiplicar(num1, num2))

        elif opcion == "dividir":
            print("Resultado:", dividir(num1, num2))

    else:
        print("Opción no válida")

File: programas/rstudio/unidad1/calculadora.r
# Función para realizar la suma de dos números
sumar <- function(a, b) {
  return(a + b)
}

# Función para realizar la resta de dos números
restar <- function(a, b) {
  return(a - b)
}

# Función para realizar la multiplicación de dos números
multiplicar <- function(a, b) {
  return(a * b)
}

# Función para realizar la división de dos números
dividir <- function(a, b) {
  if (b == 0) {
    return("Error: No se puede dividir por cero")
  }
  return(a / b)
}

# Menú de la calculadora
while (TRUE) {
  cat("Opciones:\n")
  cat("1. Sumar\n")
  cat("2. Restar\n")
  cat("3. Multiplicar\n")
  cat("4. Dividir\n")
  cat("5. Salir\n")
  
  opcion <- as.numeric(readline("Selecciona una opción: "))
  
  if (opcion == 5) {
    break
  }
  
  if (opcion >= 1 && opcion <= 4) {
    num1 <- as.numeric(readline("Ingresa el primer número: "))
    num2 <- as.numeric(readline("Ingresa el segundo número: "))
    
    if (opcion == 1) {
      resultado <- sumar(num1, num2)
    } else if (opcion == 2) {
      resultado <- restar(num1, num2)
    } else if (opcion == 3) {
      resultado <- multiplicar(num1, num2)
    } else if (opcion == 4) {
      resultado <- dividir(num1, num2)
    }
    
    cat("Resultado: ", Resultado, "\n")
  } else {
    cat("Opción no válida\n")
  }
}